crates/ide-completion/src/completions/type.rs

```diff
@@ -31,6 +31,11 @@ pub(crate) fn complete_type_path(
             ScopeDef::ImplSelfType(_) => location.complete_self_type(),
             // Don't suggest attribute macros and derives.
             ScopeDef::ModuleDef(Macro(mac)) => mac.is_fn_like(ctx.db),
+            ScopeDef::ModuleDef(Trait(_) | Module(_))
+                if matches!(location, TypeLocation::ImplTrait) =>
+            {
+                true
+            }
```

Comment:
```suggestion
            ScopeDef::ModuleDef(Trait(_) | Module(_)) => matches!(location, TypeLocation::ImplTrait),
```

Reply:
I've used match guard because I want it to fall to this match case [here](https://github.com/dfireBird/rust-analyzer/blob/0e47befaf3cdf30fcf851fd34d4d6fdb4d274382/crates/ide-completion/src/completions/type.rs#L40), if it's not `ImplTrait` location.

Reply:
Oh right, don't mind me then.